### DISCUSSION TITLE
v0.10.0-BETA.1

### DISCUSSION THREAD

**Discussion by adultmm** [adultmm/AdultMediaManager#22] [General]
**I have implemented the possibility to use "meta-scrapers" like ThePornDB or Data18 for scraping video info. It required a lot of code refactoring, so please, be patient if you found an error. :)  Also added in-memory cache possibility to speed up the scraping and 18 new sites.**

- **ADD: "meta-scraper" functionality to Video scraping (possibility to download video information from ThePornDB or Data18.com)**
- **ADD: ThePornDB meta-scraper**
- **ADD: Data18 meta-scraper**
 
- ADD: scrape actors' images from IAFD.com as well

- ADD: TukTuk Patrol scraper
- ADD: FantasyMassage.com scraper (NuruMassage.com, TrickySpa.com, Massage-Parlor.com and six more sites)
- ADD: WoodmanCastingX scraper
- ADD: JaysPOV.net scraper
- ADD: X-Art.com scraper
- ADD: Hegre.com scaper
- ADD: FrolicMe.com scraper 
- ADD: XConfessions scraper
- ADD: Slayed.com site support to VixenGroup scraper
- ADD: BackroomCastingCouch scraper now supports actors' images 
- ADD: ExCoGi scraper now supports actors' images
- ADD: ExCoGi scraper now supports "ecg" abbreviations in filenames
- ADD: Microsoft.Extensions.Caching.Memory library for in-process caching
 
- MOD: better error handling in WebClient when switching to Tor proxy
- MOD: video scoring now handles the "and" ~ "," ~ "&" situations
- MOD: AnalVids scraper now matches LegalPorno files
- MOD: replaces Chr(3) to space in movie info (illegal character in XML)
- MOD: decrease the score of RealityKings search results without detailed info (auto scraping now can select the better result)
- MOD: removed the www prefix from ExCoGi URLs
- MOD: NewSensation scraper now uses ImgBox to store secured images
- MOD: ImageCache now also uses an in-memory cache
- MOD: Scrapers now use cache for downloaded movie info

- FIX: TrueAnal.com didn't find some videos
- FIX: duplicated actors were scraped from GammaEntertainment sites (EvilAngel, 21Sextury, PureTaboo, etc.)
- FIX: NaughtyAmerica site changed - no runtime found
- FIX: ImgBox referrer support
- FIX: the loading panel may have stuck on the VideoSearchResult page 
 

<hr /><em>This discussion was created from the release <a href='https://github.com/adultmm/AdultMediaManager/releases/tag/v0.10.0-BETA.1'>v0.10.0-BETA.1</a>.</em>

**Comment by godfathertulpe** [comment]
Great update!

**Comment by adultmm** [comment]
Thanks!

**Comment by chaitrakiran2010** [comment]
Wonderful update !!!!!

**Comment by chaitrakiran2010** [comment]
Extraordinary update.....almost 90% of porn sites covered by ThePornDB, which is implemented in v0.10.0

**Comment by adultmm** [comment]
Thank you, Guys!
